crates/context/src/tx.rs

```diff
@@ -252,6 +252,176 @@ impl Transaction for TxEnv {
     // }
 }
 
+/// Builder for constructing [`TxEnv`] instances
+#[derive(Default, Debug)]
+pub struct TxEnvBuilder {
+    tx_type: u8,
```

Comment:
Lets make this `Option<u8>` and on `build` when you create TxEnv call `derive_tx_type`